libcontainerd/client.go

```diff
@@ -0,0 +1,310 @@
+package libcontainerd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sync"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	containerd "github.com/docker/containerd/api/grpc/types"
+	"github.com/opencontainers/specs"
+	"golang.org/x/net/context"
+)
+
+// Client privides access to containerd features.
+type Client interface {
+	CreateContainer
+	Signal(id string, sig int) error
+	AddProcess(id, processID string, process specs.Process) error
+	Resize(id, processID string, width, height int) error
+	Pause(id string) error
+	Resume(id string) error
+	Restore(id string) error
+	Stats(id string) (*containerd.StatsResponse, error)
+	GetPidsForContainer(id string) ([]int, error)
+}
+
+type client struct {
+	sync.Mutex
+	arrayLock  sync.RWMutex
+	locks      map[string]*sync.Mutex
```

Comment:
It would be nice to have comments explaining how these various mutexes are used.